File: src/main/java/phoenix/utils/BlockPosUtils.java
package phoenix.utils;

import net.minecraft.util.math.BlockPos;

import java.util.Collection;

public class BlockPosUtils
{
    public static double distanceTo(BlockPos first, BlockPos second)
    {
        return Math.sqrt((first.getX() - second.getX()) * (first.getX() - second.getX()) +
                (first.getX() - second.getY()) * (first.getY() - second.getY()) +
                (first.getX() - second.getZ()) * (first.getZ() - second.getZ()));
    }

    public static boolean isNear(BlockPos pos, Collection<BlockPos> poses, int range)
    {
        for(BlockPos pos1 : poses)
            if(distanceTo(pos, pos1) < range)
                return false;
        return true;
    }
}

File: src/main/java/phoenix/utils/BlockPosUtils.kt
package phoenix.utils

import net.minecraft.util.math.BlockPos
import kotlin.math.sqrt

object BlockPosUtils
{
    operator fun BlockPos.minus(second: BlockPos) : Double
    {
        return sqrt(((x - second.x) * (x - second.x) + (x - second.y) * (y - second.y) + (x - second.z) * (z - second.z)).toDouble())
    }

    fun isNear(pos: BlockPos, poses: Collection<BlockPos>, range: Int): Boolean
    {
        for (pos1 in poses) if (pos - pos1 < range) return false
        return true
    }
}